packages/react/src/components/Chat/ChatMessage.tsx

```diff
@@ -113,116 +121,118 @@ export interface ChatMessageProps
   unstable_overflow?: boolean
 }
 
-export interface ChatMessageState {
-  focused: boolean
-  messageNode: HTMLElement
-}
-
-class ChatMessage extends UIComponent<WithAsProp<ChatMessageProps>, ChatMessageState> {
-  static className = 'ui-chat__message'
-
-  static create: ShorthandFactory<ChatMessageProps>
-
-  static slotClassNames: ChatMessageSlotClassNames
-
-  static displayName = 'ChatMessage'
-
-  static __isChatMessage = true
-
-  static isTypeOfElement = element => _.get(element, `type.__isChatMessage`)
-
-  static propTypes = {
-    ...commonPropTypes.createCommon({ content: 'shorthand' }),
-    actionMenu: customPropTypes.itemShorthand,
-    attached: PropTypes.oneOfType([PropTypes.bool, PropTypes.oneOf(['top', 'bottom'])]),
-    author: customPropTypes.itemShorthand,
-    badge: customPropTypes.itemShorthand,
-    badgePosition: PropTypes.oneOf(['start', 'end']),
-    mine: PropTypes.bool,
-    timestamp: customPropTypes.itemShorthand,
-    onBlur: PropTypes.func,
-    onFocus: PropTypes.func,
-    onMouseEnter: PropTypes.func,
-    positionActionMenu: PropTypes.bool,
-    reactionGroup: PropTypes.oneOfType([
-      customPropTypes.collectionShorthand,
-      customPropTypes.itemShorthand,
-    ]),
-    reactionGroupPosition: PropTypes.oneOf(['start', 'end']),
-    unstable_overflow: PropTypes.bool,
-  }
-
-  static defaultProps = {
-    accessibility: chatMessageBehavior,
-    as: 'div',
-    badgePosition: 'end',
-    positionActionMenu: true,
-    reactionGroupPosition: 'start',
-  }
-
-  updateActionsMenuPosition = () => {}
-
-  state = {
-    focused: false,
-    messageNode: null,
-  }
-
-  menuRef = React.createRef<HTMLElement>()
+export type ChatMessageStylesProps = Pick<
```

Comment:
It's good that we have these typings now on the component level, but let's make sure that we move the others too :)